src/vulkan/command_buffer.cc

```diff
@@ -120,6 +122,9 @@ Result CommandBuffer::SubmitAndReset(uint32_t timeout_ms) {
   if (device_->GetPtrs()->vkResetCommandBuffer(command_, 0) != VK_SUCCESS)
     return Result("Vulkan::Calling vkResetCommandBuffer Fail");
 
+  if((timeout_ms == EngineData::DEFAULT_IMEOUT) || (timeout_ms == uint32_t(-1)))
```

Comment:
Why do we care about the timeout value? If the fence is already over, shouldn't we call this if `vkWaitForFences` returns `VK_SUCCESS`? We should also comment why we are calling it because it is not obvious this signals the performance layers.

Reply:
Good call, I added the comment from the PR immediately before the call.
Also this should happen after all the checks for failures so we should be in the VK_SUCCESS case.

Ok i will change the reliance on timeout.